common/source/docs/common-fettec-onewire.rst

```diff
@@ -0,0 +1,101 @@
+==============
+FETtec OneWire
+==============
+
+This is a half duplex serial protocol with a 2Mbit/s Baudrate created by Felix Niessen (former Flyduino KISS developer) from `FETtec <https://fettec.net/>`_.
+You can change many more settings of the ESC over OneWire than over DShot.
+It also has a better cyclic redundancy check (CRC) than Dshot offers.
+CRC checks if there are any signal errors that occur in the communication between autopilot and ESC; and between ESC and autopilot.
+It supports ESC telemetry, so information from the ESC status can be sent back to the autopilot:
+
+- Motor rotations per minute (RPM)
+- Input voltage (V)
+- Current draw (A)
+- Power consumption (W)
+- Temperature (°C)
+
+This information allows the autopilot to:
+
+- dynamicaly change the center frequency of the notch filters used to reduce frame vibration noise in the gyros
+- log the status of each ESC to the SDCard or internal Flash, for post flight analysis
+- send the status of each ESC to the Ground Station or companion computer for real-time monitoring
+
+Connecting
+==========
+
+The protocol supports up-to 25 ESCs, but the ArduPilot implementation of it limits this to 12.
+You can connect all ESCs to a single serial port using only a single signal wire from the autopilot's UART TX pin to the ESC's TLM pin and one GND wire.
+If you are using 4in1 ESCs you can connect up-to 3 of them in this fashion.
+
+TODO: add a figure here.
+
+FullDuplex with Diode
+=====================
+If halfduplex is not available or working there is a workaround with a diode using the normal full duplex.
+This is also working with other "non-halfduplex" Controller like ESP32.
+
+The TLM-Pin (ESC) and RX (FC) connects to the anode (black side) of the diode.
+TX (FC) connects to cathode (white striped side) of the diode.
+
+ (TX --|<-- RX, TLM)
+
+Connection and Halfduplex with STM
```

Comment:
@teezeefpv please comment on this

Reply:
this section need to be removed as it will only confuse people that just want to plug their ESC.
And the part of the code means nothing without knowing what environment we are using (FreeRTOS? CubeMX?)